### ISSUE TITLE
Module access issue with jLink

### ISSUE THREAD

**Issue by 0rbianta** [oshi/oshi#2501]
Exception in thread "JavaFX Application Thread" java.lang.IllegalAccessError: class oshi.util.platform.windows.WmiQueryHandler (in module com.testapp.merged.module) cannot access class org.slf4j.LoggerFactory (in module org.slf4j) because module com.testapp.merged.module does not read module org.slf4j

I tried to run the shadow/fat jar version of my application and it works fine but when jLink is used, it gives the exception given above. I don't have such a thing like com.testapp.merged.module on my app. Also other dependenices worked fine so I think there is something with oshi. What can I do or what is causing the problem?

**Comment by dbwiddis** [comment]
I'm not completely clear on the issue (and am not a JPMS expert) but have a few questions:
1. When you say you're using shadow/fat jar, which jar is it?  One produced by OSHI's build?
2. Are you using the `oshi-core-java11` jar which includes a [module descriptor](https://github.com/oshi/oshi/blob/master/oshi-core-java11/src/main/java/module-info.java)
3. Which version of slf4j are you using?  Does it have a module descriptor?
4. Does your project have a module descriptor and does it include `requires com.github.oshi`?

Digging around I found https://github.com/beryx/badass-jlink-plugin/issues/127 which indicates you may need something like 
```json
    mergedModule {
        additive = true
        requires("org.slf4j")
    }
```

If that doesn't work, the error is relatively common and can usually be worked around with `--add-exports` command line switch.  I'm guessing here, but try variations on:
```
--add-exports org.slf4j/org.slf4j=com.testapp.merged.module
```
or even more broad
```
--add-exports org.slf4j/org.slf4j=ALL-UNNAMED
```

**Comment by 0rbianta** [comment]
1. **When you say you're using shadow/fat jar, which jar is it? One produced by OSHI's build?**  
       The fat jar in question pertains to my own application's build and is not produced by OSHI's build process.

2. **Are you using the oshi-core-java11 jar which includes a [module descriptor]?(https://github.com/oshi/oshi/blob/master/oshi-core-java11/src/main/java/module-info.java)**  
   Initially, I wasn't using the oshi-core-java11 jar because I'm on Java 17 LTS and assumed that the artifact was only for Java 11. However, I've come to realize that this artifact is for Java 11+, and it includes the necessary module descriptor, making it compatible with my jLink build.

3. **Which version of slf4j are you using? Does it have a module descriptor?**  
   I'm not directly using slf4j in my project; it's a dependency of OSHI. The issue I faced was OSHI being unable to access slf4j, but this was resolved when I switched to using the module-info-included version of OSHI.

4. **Does your project have a module descriptor and does it include requires com.github.oshi?**  
   Yes, my project includes a module-info.java file and it specifies `requires com.github.oshi`.

In conclusion, the root of the issue was the utilization of an incorrect dependency. Upon switching to the oshi-core-java11 artifact, which is compatible with Java 11 and higher versions and includes Java's module system, the problem was effectively resolved.

**Comment by dbwiddis** [comment]
Awesome.  Glad you got it working and all I had to do was ask a bunch of questions. :)

> I wasn't using the oshi-core-java11 jar because I'm on Java 17 LTS and assumed that the artifact was only for Java 11. 

Thanks for the beedback. I agree it could be better documented.
